### PR TITLE
[colmap] Update to latest version from 2023-10-01

### PR DESCRIPTION
**Pull request by ahojnnes**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Cheney-W** [comment]
Could you please solve the CI test failures?
Failures logs: https://dev.azure.com/vcpkg/public/_build/results?buildId=93075&view=artifacts&pathAsName=false&type=publishedArtifacts

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -225,6 +225,8 @@ cnats:arm-neon-android=fail
 cnats:arm64-android=fail
 cnats:x64-android=fail
 coin-or-ipopt:x64-linux=fail
+colmap:x64-windows-static=fail
+colmap:x64-windows-static-md=fail
```
Comment (Cheney-W) [inline scripts/ci.baseline.txt:229]: Does this mean that this port no longer supports static? If so, please add the support field in vcpkg.json file.

**Comment by Cheney-W** [comment]
I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (ahojnnes) [reply scripts/ci.baseline.txt:221]: I hope this is correct?

**Comment by ahojnnes** [comment]
@Cheney-W PTAL, thank you.

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (Cheney-W) [reply scripts/ci.baseline.txt:221]: I saw you have added `windows & !static` into the support, please remove above records from ci.baseline.txt.

Comment (ahojnnes) [reply scripts/ci.baseline.txt]: OK, thanks, wasn't sure. Done.

File: 9c704e89ff0a80e797725e112011f9f69e5aa28f.tar.gz
Comment (Cheney-W) [inline 9c704e89ff0a80e797725e112011f9f69e5aa28f.tar.gz:1]: What's this? I think this should be removed.

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: 9c704e89ff0a80e797725e112011f9f69e5aa28f.tar.gz
Comment (ahojnnes) [reply 9c704e89ff0a80e797725e112011f9f69e5aa28f.tar.gz:1]: Accidentally committed. Apologies.

**Review by Cheney-W — APPROVED** [review]

**Review by Cheney-W — APPROVED** [review]

File: ports/colmap/fix-dependencies.patch [hunk 2]
```diff
@@ -1,23 +0,0 @@
---- CMakeLists.txt.orig	2023-03-12 19:26:41.000000000 +0100
-+++ CMakeLists.txt	2023-03-27 11:27:43.840624719 +0200
-@@ -99,14 +99,17 @@
- 
- find_package(Eigen3 REQUIRED)
- 
--find_package(FreeImage REQUIRED)
```
Comment (BillyONeal) [inline ports/colmap/fix-dependencies.patch:7]: I confirmed that these changes are obsoleted by https://github.com/colmap/colmap/pull/2116/

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**